tsconfig.json

```diff
@@ -1,12 +1,16 @@
 {
   "compilerOptions": {
-    "target": "ES5",
-    "module": "commonjs",
+    "target": "ES2016",
```

Comment:
@shortaflip 
Hi, thanks for the comment. Here I will explain some of my doubts to know also how you think about how we should act.

I understand the need to be compatible with old browsers, but even using the library so "naked" it is not possible to cover a cross-browser target (necessary in the case of a CMP) without adding some polyfills.
Because there are some holes. I think that in this library it is quite difficult to define support for certain browsers. 
I also think that in fact, it is necessary to leave the freedom to the developer of a CMP to choose the target as it also depends on the type of website you are going to add this code to.

But I understand the need to maybe expose the library in both `ESM` and `commonjs`.
But I see that the documentation uses all examples showing the use of the library through `import` statement, this however means that whoever uses the library is already using ESM (maybe).

Do you think it would be better to export both `ESM` and `commonjs` library anyway?

Thank you in advance 😜

Reply:
@shortaflip 
Hi 😁 
Being that there might be a possibility that a developer would want to use `commonjs` rather than `esm`. 
I updated the PR by updating the build flow for library generation in both `esm` and `commonjs` module types.

With these changes in practice, the build will create 2 folders inside `lib`. One called `cjs` stands for `commonjs` using the default tsconfig configuration that already exists in the project. And another folder called `mjs` which uses the configuration needed to export modules via `ESM`.

Could you deploy this branch perhaps under an 'alpha' tag at least to test the actual output? Because locally I'm using Symlinks but the "npm publish" command does something else and I'd like to try a release taken from the npm registry directly.

Thank you in advance 🙏

Reply:
Hey @Marco-Prontera, sorry I didn't communicate this well.  What I want to make sure, is that this update doesn't break any CMP's in production that use this library via `require`.

So given your proposal, there are two questions:
1. Does outputting both `cjs` and `esm` mean that CMP's (especially those using `require`) might potentially break?
2. Does outputting both `cjs` and `esm` mean extra maintenance for this library in the future?

Reply:
Hi @shortaflip, I'll answer your questions:

1. To answer you with certainty it would be advisable to publish a beta or alpha tagged release. This allows us to verify only this branch without affecting anyone. I still read the npm documentation to correctly configure the `require` and `import` of those who use the library, but publishing a beta or alpha doesn't hurt anyone 😜

2. The output is handled directly by the typescript decoding process (`tsc`) via two configurations that meet the requirements of `cjs` and `esm`. So once the setup is done there is no need for extra maintenance.

However, if this change is good but is a breaking change at that point a major can be released. This is to be sure that whoever updates the library will check everything.

Let me know ;)

Reply:
Ok that sounds good, I'll tag you an alpha-release this Wednesday.  If there is a breaking change though Marco, this update will have to wait.  As there is potential for the specifications of this library to change when the IAB finalizes things regarding the Belgium Decision.

Reply:
`yarn workspaces run build` doesn't produce any errors from me either, `yarn test` does.

Reply:
<img width="868" alt="Schermata 2022-05-04 alle 18 41 39" src="https://user-images.githubusercontent.com/50167854/166729622-7229e03e-f22f-40cd-b83f-de47dba81197.png">
Same all tests pass

Reply:
@Marco-Prontera So I have been trying to get you an alpha build published on npm but the publish workflow of this library was written in a very restrictive way.
(only publishes from master, pre-release -> release cycle only, and other checks that are stopping me from doing an alpha build)

What I am going to have to do is the following:
* merge in both of your PRs
* publish a pre-release on npm that contains the current changes (including yours)

The pre-release will be tagged as `next`.

You'll be able to test your changes from that pre-release build.  We will then be in a PR -> pre-release -> PR -> pre-release cycle until your changes are good.  If we find out that your changes just won't work for our usecase, I can revert the PRs on the master branch.

If you are ok with this, let me know.

Reply:
It's ok for me. If you want I made you a bash script on the fly to publish an alpha version. Maybe create a file inside the bin folder and put this code in it so you can try:

`#!/bin/bash

ROOT=$(git rev-parse --show-toplevel)

cd $ROOT

for pkg in ./modules/*; do
  if [[ -d $pkg ]]; then

    echo
    cd $pkg
    echo "Changing $pkg/package.json to version $npm_package_version"
    json -I -f package.json -e "this.version=\"$npm_package_version\""
    cd -
  fi
done

sed -i '' 's/\(\@iabtcf\/.*\)":.*"[0-9]*\..*"/\1": "'$npm_package_version'"/g' ./modules/**/package.json

yarn build

for pkg in ./modules/*; do
  if [[ -d $pkg ]]; then
    yarn publish --access public --tag alpha --new-version $npm_package_version $pkg
  fi
done`

if it doesn't work then proceed to do as you said. 😉

Reply:
Thanks aha but it didn't work.

Reply:
@shortaflip 
I saw the problem on the configuration of circleci. In practice there is node: 10, instead now you can use the new lts of node. I pushed a commit here to fix the problem

Reply:
Thank you for trying with that script Marco but it didn't work.  I am going to merge this in and start the pre-release plan.

Reply:
@Marco-Prontera pre-release v1.5.0-0 is up on npm with `next` tag.

Reply:
@shortaflip 
Perfect! I start to take the tests (where I am it is late in the evening, so I ask you to give me some time to take all the tests)
Can you tell me a deadline, please?

Reply:
I have a good amount of internal work and there isn't an expectation from the iab on anything that is going out with this PR, so no hard deadline.  Maybe a couple of weeks or more, no rush, getting it tested thoroughly should be the goal.

Reply:
Perfect, I do the tests with both `cjs` and `esm`. I take out a list of things to fix and then recreate a PR, with the description and explanation of everything.